src/main/scala/ai/privado/languageEngine/ruby/tagger/collection/CollectionTagger.scala

```diff
@@ -0,0 +1,132 @@
+package ai.privado.languageEngine.ruby.tagger.collection
+
+import ai.privado.cache.RuleCache
+import ai.privado.model.{Constants, InternalTag, RuleInfo}
+import ai.privado.tagger.PrivadoParallelCpgPass
+import ai.privado.utility.Utilities.*
+import io.shiftleft.codepropertygraph.generated.Cpg
+import io.shiftleft.codepropertygraph.generated.nodes.{Call, Method}
+import io.shiftleft.semanticcpg.language.*
+import org.slf4j.LoggerFactory
+
+import scala.collection.mutable
+import scala.util.{Failure, Success, Try}
+
+class CollectionTagger(cpg: Cpg, ruleCache: RuleCache) extends PrivadoParallelCpgPass[RuleInfo](cpg) {
+  private val logger       = LoggerFactory.getLogger(this.getClass)
+  private val methodUrlMap = mutable.HashMap[Long, String]()
+  private val classUrlMap  = mutable.HashMap[Long, String]()
+
+  override def generateParts(): Array[RuleInfo] =
+    ruleCache.getRule.collections.filter(_.catLevelTwo == Constants.default).toArray
+
+  override def runOnPart(builder: DiffGraphBuilder, collectionRuleInfo: RuleInfo): Unit = {
+
+    val collectionMethods = cpg
+      .call("get|post|put|patch|delete")
+      .where(_.file.name(".*routes.rb"))
+      .filter(_.argument.isCall.code("to.*").argument.isLiteral.nonEmpty)
+      .l
+
+    /*
+    creating map having key as method Id which is getting invoked by collection and value as url
+     */
+    val collectionMethodsCache = collectionMethods
+      .map { m =>
+        //        sample route -> get '/hotels/new', to: 'hotels#new'
+        // TODO: check scenarios involving single, double and no quote
+        val targetCollectionUrl = m.argument.isCall.code("to:.*").argument.isLiteral.code.l
+        if (!targetCollectionUrl.isEmpty) {
+          val routeMethodCall: String = targetCollectionUrl.head.stripPrefix("\"").stripSuffix("\"")
+          val routeSeparatedStrings   = routeMethodCall.split("#")
+          val methodName              = routeSeparatedStrings(1)
+          val fileName                = ".*" + routeSeparatedStrings(0) + ".*"
```

Comment:
Better to always use `headOption` instead of `using directly index when looking for 1st object in a array/list`